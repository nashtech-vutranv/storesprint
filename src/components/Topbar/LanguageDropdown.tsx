import {useState, useEffect} from 'react'
import i18next from 'i18next'
import {useTranslation} from 'react-i18next'
import {Link} from 'react-router-dom'
import {Dropdown} from 'react-bootstrap'
import {listLanguageSelected} from '../../constants/'
import {LanguageKey, LanguageValue} from '../../interface/language'
import {getCurrentLanguageValue} from '../../helpers/language'

const enFlag = require('./flags/uk.jpg')
const frFlag = require('./flags/french.jpg')

const LanguageDropdown = () => {
  const {t} = useTranslation()

  const [dropdownOpen, setDropdownOpen] = useState<boolean>(false)

  const [currentLanguage, setCurrentLanguage] = useState<string>(
    getCurrentLanguageValue()
  )

  const handleSelectLanguage = (language: {
    id: number
    value: string
    key: LanguageKey
  }) => {
    i18next.changeLanguage(language.key)
    toggleDropdown()
  }

  const toggleDropdown = () => {
    setDropdownOpen(!dropdownOpen)
  }

  useEffect(() => {
    setCurrentLanguage(getCurrentLanguageValue())
  }, [localStorage.getItem('i18nextLng')])

  return (
    <Dropdown show={dropdownOpen} onToggle={toggleDropdown}>
      <Dropdown.Toggle
        variant='link'
        id='dropdown-languages'
        onClick={toggleDropdown}
        className='nav-link dropdown-toggle arrow-none shadow-none'
      >
        <img
          src={
            getCurrentLanguageValue() === LanguageValue.english
              ? enFlag
              : frFlag
          }
          alt={getCurrentLanguageValue()}
          className='me-1'
          height='12'
          width='20'
        />{' '}
        <span className='align-middle d-none d-sm-inline-block'>
          {currentLanguage}
        </span>
        <i className='mdi mdi-chevron-down d-none d-sm-inline-block align-middle'></i>
      </Dropdown.Toggle>
      <Dropdown.Menu className='dropdown-menu dropdown-menu-start dropdown-menu-animated topbar-dropdown-menu'>
        <div onClick={toggleDropdown}>
          {listLanguageSelected.map((lang) => {
            const translatedLanguage = {
              ...lang,
              value:
                lang.value === LanguageValue.english
                  ? t('header_language_english')
                  : t('header_language_french'),
            }
            return (
              <Link
                onClick={() => handleSelectLanguage(translatedLanguage)}
                to='#'
                className='dropdown-item notify-item'
                key={translatedLanguage.key}
              >
                <img
                  src={lang.key === LanguageKey.english ? enFlag : frFlag}
                  alt={getCurrentLanguageValue()}
                  className='me-1'
                  height='12'
                  width='20'
                />{' '}
                <span className='align-middle'>{translatedLanguage.value}</span>
              </Link>
            )
          })}
        </div>
      </Dropdown.Menu>
    </Dropdown>
  )
}

export default LanguageDropdown
