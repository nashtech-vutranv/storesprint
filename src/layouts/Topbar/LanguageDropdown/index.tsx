import {Link} from 'react-router-dom'
import {Dropdown} from 'react-bootstrap'
import i18next from 'i18next'
import {useTranslation} from 'react-i18next'
import {useEffect, useState} from 'react'
import {useToggle} from '../../../hooks'
import {getCurrentLanguageValue} from '../../../helpers/language'
import {LanguageKey, LanguageValue} from '../../../interface/language'
import {listLanguageSelected} from '../../../constants'
import enFlag from './flags/uk.svg'
import frFlag from './flags/fr.svg'

const LanguageDropdown = () => {
  const {t} = useTranslation()
  const [currentLanguage, setCurrentLanguage] = useState<string>(
    getCurrentLanguageValue()
  )
  const [isOpen, toggleDropdown] = useToggle()

  const handleSelectLanguage = (language: {
    id: number
    value: string
    key: LanguageKey
  }) => {
    i18next.changeLanguage(language.key)
    toggleDropdown()
  }
  const selectedLanguage = localStorage.getItem('i18nextLng')
  useEffect(() => {
    setCurrentLanguage(getCurrentLanguageValue())
  }, [selectedLanguage])

  return (
    <Dropdown show={isOpen} onToggle={toggleDropdown}>
      <Dropdown.Toggle
        variant='link'
        onClick={toggleDropdown}
        className='nav-link dropdown-toggle arrow-none shadow-none'
      >
        <img
          src={
            getCurrentLanguageValue() === LanguageValue.english
              ? enFlag
              : frFlag
          }
          alt={currentLanguage}
          className='me-0 me-sm-1'
          height='12'
          width='20'
        />{' '}
        <span className='align-middle d-none d-sm-inline-block'>
          {currentLanguage}
        </span>
        <i className='mdi mdi-chevron-down d-none d-sm-inline-block align-middle'></i>
      </Dropdown.Toggle>
      <Dropdown.Menu
        align={'end'}
        className='dropdown-menu-animated topbar-dropdown-menu'
      >
        <div onClick={toggleDropdown}>
          {listLanguageSelected.map((lang) => {
            const translatedLanguage = {
              ...lang,
              value:
                lang.value === LanguageValue.english
                  ? t('header_language_english')
                  : t('header_language_french'),
            }
            return (
              <Link
                onClick={() => handleSelectLanguage(translatedLanguage)}
                to='#'
                className='dropdown-item notify-item'
                key={translatedLanguage.key}
              >
                <img
                  src={lang.key === LanguageKey.english ? enFlag : frFlag}
                  alt={getCurrentLanguageValue()}
                  className='me-1'
                  height='12'
                  width='20'
                />{' '}
                <span className='align-middle'>{translatedLanguage.value}</span>
              </Link>
            )
          })}
        </div>
      </Dropdown.Menu>
    </Dropdown>
  )
}

export default LanguageDropdown
