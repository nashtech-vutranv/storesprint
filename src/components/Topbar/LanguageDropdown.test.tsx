import {render, waitFor} from '@testing-library/react'
import '@testing-library/jest-dom'
import {MemoryRouter} from 'react-router-dom'
import LanguageDropdown from './LanguageDropdown'

test('renders LanguageDropdown', async () => {
  const tree = render(
    <MemoryRouter>
      <LanguageDropdown />
    </MemoryRouter>
  )

  await waitFor(() => {
    expect(tree).toMatchSnapshot()
  })
})
