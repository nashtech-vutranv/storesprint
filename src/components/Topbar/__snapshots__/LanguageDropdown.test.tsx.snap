// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LanguageDropdown 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="dropdown"
      >
        <button
          aria-expanded="false"
          class="nav-link dropdown-toggle arrow-none shadow-none dropdown-toggle btn btn-link"
          id="dropdown-languages"
          type="button"
        >
          <img
            alt="English"
            class="me-1"
            height="12"
            src="[object Object]"
            width="20"
          />
           
          <span
            class="align-middle d-none d-sm-inline-block"
          >
            English
          </span>
          <i
            class="mdi mdi-chevron-down d-none d-sm-inline-block align-middle"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="dropdown"
    >
      <button
        aria-expanded="false"
        class="nav-link dropdown-toggle arrow-none shadow-none dropdown-toggle btn btn-link"
        id="dropdown-languages"
        type="button"
      >
        <img
          alt="English"
          class="me-1"
          height="12"
          src="[object Object]"
          width="20"
        />
         
        <span
          class="align-middle d-none d-sm-inline-block"
        >
          English
        </span>
        <i
          class="mdi mdi-chevron-down d-none d-sm-inline-block align-middle"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
